src/witness/memory.rs

```diff
@@ -186,7 +186,7 @@ impl SafeMemory {
     /// * `ptr` - The memory address where the field element will be written.
     /// * `fr` - The [`U256`] field element to write.
     fn write_short(&mut self, store: &impl AsStoreRef, ptr: usize, fr: U256) -> Result<()> {
-        let num = fr.to_words()[0] as u32;
+        let num = fr.as_limbs()[0] as u32; // wtf is happening
```

Comment:
what's the issue here?

Reply:
I thought there might be a bug here, but after the tests passed and the constraints were satisfied, the behavior is correct